examples/sha256.rs

```diff
@@ -0,0 +1,235 @@
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    let num_of_64_bytes = args[1].parse::<usize>().unwrap();
+    let expect = hex::decode(args[2].parse::<String>().unwrap()).unwrap();
+    let _setup_only = args[3].parse::<bool>().unwrap();
+
+    let input_size = 64 * num_of_64_bytes;
+    let _input_str = vec![0u8; input_size];
+
+    let store = &mut Store::<Fr>::new();
+    let mut lang = Lang::<Fr, Sha256Coproc<Fr>>::new();
+    let sym_str = format!(".sha256.hash-{}-zero-bytes", input_size);
+    let name = Sym::new(sym_str.clone());
+    let coprocessor: Sha256Coprocessor<Fr> = Sha256Coprocessor::<Fr>::new(input_size);
+    let coproc = Sha256Coproc::SC(coprocessor);
+
+    lang.add_coprocessor(name, coproc, store);
+
+    let coproc_expr = format!("({})", sym_str);
+
+    let mut u: [u64; 4] = [0u64; 4];
+
+    for i in 0..4 {
+        u[i] = u64::from_be_bytes(expect[(i * 8)..(i + 1) * 8].try_into().unwrap())
+    }
+    let result_expr = format!("({}u64 {}u64 {}u64 {}u64)", u[0], u[1], u[2], u[3]);
+
+    let expr = format!("(emit (eq {coproc_expr} (quote {result_expr})))");
+    let ptr = store.read(&expr).unwrap();
+
+    let nova_prover = NovaProver::<Fr, Sha256Coproc<Fr>>::new(REDUCTION_COUNT, lang.clone());
+    let lang_rc = Arc::new(lang);
+
+    println!("Setting up public parameters...");
+
+    let pp_start = Instant::now();
+    let pp = public_params::<Sha256Coproc<Fr>>(REDUCTION_COUNT, lang_rc.clone()).unwrap();
+    let pp_end = pp_start.elapsed();
+
+    println!("Public parameters took {:?}", pp_end);
+
+    println!("Beginning proof step...");
+
+    let proof_start = Instant::now();
+    let (proof, z0, zi, num_steps) = nova_prover
+        .evaluate_and_prove(&pp, ptr, empty_sym_env(store), store, 10000, lang_rc)
+        .unwrap();
+    let proof_end = proof_start.elapsed();
+
+    println!("Proofs took {:?}", proof_end);
+
+    println!("Verifying proof...");
+
+    let verify_start = Instant::now();
+    let res = proof.verify(&pp, num_steps, z0, &zi).unwrap();
+    let verify_end = verify_start.elapsed();
+
+    println!("Verify took {:?}", verify_end);
+
+    if res {
+        println!(
+            "Congratulations! You proved and verified a SHA256 hash calculation in {:?} time!",
+            pp_end + proof_end + verify_end
+        );
+    }
+}
+
+fn make_u64_from_bits<F, CS>(
+    mut cs: CS,
+    bits: &[Boolean],
+) -> Result<AllocatedPtr<F>, SynthesisError>
+where
+    F: LurkField,
+    CS: ConstraintSystem<F>,
+{
+    let mut num = BNum::<F>::zero();
+    let mut coeff = F::one();
+
+    for bit in bits {
+        num = num.add_bool_with_coeff(CS::one(), bit, coeff);
+
+        coeff = coeff.double();
+    }
+
+    let allocated_num = AllocatedNum::alloc(&mut cs.namespace(|| "chunk"), || {
+        num.get_value().ok_or(SynthesisError::AssignmentMissing)
+    })?;
+    // num * 1 = input
```

Comment:
We have `enforce_pack` [here](https://github.com/lurk-lab/lurk-rs/blob/master/src/circuit/gadgets/constraints.rs#L117), which seems to do a similar work. Should we use it here?